docs/src/operate/lifecycle/deregister.md

```diff
@@ -0,0 +1,106 @@
+For e.g, for a child device `child01`, the registration message is stored on the `te/device/child01//` topic,
+its twin data is stored across several `te/device/child01///twin/<twin-data-type>` topics,
+its command metadata stored across several `te/device/child01///cmd/<cmd-type>` topics and so on.
+
+On top of that, when a device has services associated with it, or has nested child devices,
+they all have their own respective registration topics and other metadata topics.
+So, de-registering a device involves de-registering itself, its metadata and
+the complete entity hierarchy that is associated with it.
+
+## Deregister a single entity
+
+De-registering a single entity, excluding its child entities, involves clearing its registration message and
+all the metadata messages associated with it from the MQTT broker.
+Even though thin-edge doesn't provide a direct API for this yet, this can be easily be done using third-party tools
+like `mosquitto_sub` that supports clearing multiple retained messages together using the `--remove-retained` option.
+
+For e.g, a child device with topic id `device/child01//` can be deregistered as follows:
+
+```sh
+mosquitto_sub --remove-retained -W 3 -t "te/device/child01///#"
+```
+
+The topic filter `te/device/child01///#` covers the device's registration message as well all of its metadata messages.
+Running the same command without the `remove-retained` option shows the list of messages that would be cleared with this command.
+The `-W 3` option is used to stop the command after a maximum timeout period of 3 seconds.
+Adjust this timeout value based on the number of entities in your deployment.
+
+:::note
+Deregistering the main device using this mechanism is not supported.
+:::
+
+Similarly, a service with topic id `device/main/service/service01` can be deregistered as follows:
+
+```sh
+mosquitto_sub --remove-retained -W 3 -t "te/device/main/service/service01/#"
+```
+
+Services of child devices can also be de-registered in a similar manner, using the child device topic id instead.
+
+:::note
+The Cumulocity mapper must be restarted for the de-registrations to take effect.
+The entities in the cloud must also be removed separately as de-registrations on the device
+are not propagated to the cloud.
+:::
+
+## Deregister a child device and its services
+
+When the topic ids of the devices and services follow the default topic scheme,
+which maintains the service topic ids (`te/device/<device-id>/service/<service-id>`)
+directly under the device topic id (`te/device/<device-id>//`) in the hierarchy,
+all the services associated to a child device can be deregistered as follows:
+
+```sh
+mosquitto_sub --remove-retained -W 3 -t "te/device/child01/service/+/#"
+```
+
+The device can also be deleted along with all of its services with a minor variation in the topic
+that includes the device as well, as follows:
+
+```sh
+mosquitto_sub --remove-retained -W 3 -t "te/device/child01/+/+/#"
```

Comment:
Can't the `#` wildcard pattern be used to delete the entity registration and the related telemetry/command topics?

```suggestion
mosquitto_sub --remove-retained -W 3 -t "te/device/child01/#"
```

Reply:
Sorry, ignore this comment...I misread the number of slashes...